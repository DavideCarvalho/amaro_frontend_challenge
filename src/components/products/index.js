// @flow
import ProductList from './product_list';
import Product from './product';
export const ProductListComponent = ProductList;
export const ProductComponent = Product;