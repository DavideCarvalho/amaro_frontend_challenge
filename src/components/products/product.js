// @flow
import React from 'react';
import type { ProductType } from '../../types/Products/product';

export default class Product extends React.Component<ProductType> {
  constructor(props: ProductType) {
    super(props);
    console.log(props);
  }
  render () {
    return (
      <h1>{this.props.name}</h1>
    )
  }
}