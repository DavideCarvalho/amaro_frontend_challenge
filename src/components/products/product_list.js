// @flow
import React from 'react';
import Product from "./product";

type ProductListContainerType = {
  productsState: ProductListType,
  getProducts: () => void
}

export default class ProductList extends React.Component<ProductListContainerType> {

  componentWillMount() {
    this.props.getProducts();
  }

  render () {
    console.log(this.props.productsState.products[0])
    return (
      <div className="columns is-multiline">{
        !this.props.productsState.products ?
        <h1>Carregando</h1> :
        //this.props.productsState.products.map(product => {
          //return (
            //<div class="is-one-quarter bd-notification">
              //<Product {...product} />
              <Product {...this.props.productsState.products[0]} />
            //</div>
          //)
        //})
      }
      </div>
    )
  }
}
