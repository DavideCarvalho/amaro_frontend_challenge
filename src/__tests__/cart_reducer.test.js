import carrinhoReducer from '../reducers/cart_reducer'
import products from '../../products.json';

let cartWithOneProduct = null;
let cartWithTwoProducts = null;

beforeEach(() => {
  cartWithOneProduct = {
    cart: [
      {
        "name": "VESTIDO TRANSPASSE BOW",
        "style": "20002605",
        "code_color": "20002605_613",
        "color_slug": "tapecaria",
        "color": "TAPE�ARIA",
        "on_sale": false,
        "regular_price": "R$ 199,90",
        "actual_price": "R$ 199,90",
        "discount_percentage": "",
        "installments": "3x R$ 66,63",
        "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002605_615_catalog_1.jpg?1460136912",
        "sizes": [
          {
            "available": false,
            "size": "PP",
            "sku": "5807_343_0_PP"
          },
          {
            "available": true,
            "size": "P",
            "sku": "5807_343_0_P"
          },
          {
            "available": true,
            "size": "M",
            "sku": "5807_343_0_M"
          },
          {
            "available": true,
            "size": "G",
            "sku": "5807_343_0_G"
          },
          {
            "available": false,
            "size": "GG",
            "sku": "5807_343_0_GG"
          }
        ]
      }
    ]
  };
  
  cartWithTwoProducts = {
    cart: [
      {
        "name": "VESTIDO TRANSPASSE BOW",
        "style": "20002605",
        "code_color": "20002605_613",
        "color_slug": "tapecaria",
        "color": "TAPE�ARIA",
        "on_sale": false,
        "regular_price": "R$ 199,90",
        "actual_price": "R$ 199,90",
        "discount_percentage": "",
        "installments": "3x R$ 66,63",
        "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002605_615_catalog_1.jpg?1460136912",
        "sizes": [
          {
            "available": false,
            "size": "PP",
            "sku": "5807_343_0_PP"
          },
          {
            "available": true,
            "size": "P",
            "sku": "5807_343_0_P"
          },
          {
            "available": true,
            "size": "M",
            "sku": "5807_343_0_M"
          },
          {
            "available": true,
            "size": "G",
            "sku": "5807_343_0_G"
          },
          {
            "available": false,
            "size": "GG",
            "sku": "5807_343_0_GG"
          }
        ]
      },
      {
        "name": "REGATA ALCINHA FOLK",
        "style": "20002570",
        "code_color": "20002570_614",
        "color_slug": "preto",
        "color": "PRETO",
        "on_sale": false,
        "regular_price": "R$ 99,90",
        "actual_price": "R$ 99,90",
        "discount_percentage": "",
        "installments": "3x R$ 33,30",
        "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002570_002_catalog_1.jpg?1459948578",
        "sizes": [
          {
            "available": true,
            "size": "PP",
            "sku": "5723_40130843_0_PP"
          },
          {
            "available": true,
            "size": "P",
            "sku": "5723_40130843_0_P"
          },
          {
            "available": true,
            "size": "M",
            "sku": "5723_40130843_0_M"
          },
          {
            "available": true,
            "size": "G",
            "sku": "5723_40130843_0_G"
          },
          {
            "available": true,
            "size": "GG",
            "sku": "5723_40130843_0_GG"
          }
        ]
      }
    ]
  };
  
  
})

test('Returns default state', () => {
  expect(carrinhoReducer(undefined, {})).toMatchSnapshot();
})

test('Adds new product', () => {
  expect(carrinhoReducer(undefined, {
    type:"ADD_PRODUCT_TO_CART",
    payload: products.products[0]
  })).toMatchSnapshot();
})

test('Add a second product', () => {
  expect(carrinhoReducer(cartWithOneProduct, {
    type:"ADD_PRODUCT_TO_CART",
    payload: products.products[1]
  })).toMatchSnapshot();
})

test('Delete a existing product', () => {
  expect(carrinhoReducer(cartWithTwoProducts, {
    type:"DELETE_PRODUCT_FROM_CART",
    payload: 1
  })).toMatchSnapshot();
})

test('Delete a non existing product should return the same state', () => {
    expect(carrinhoReducer(cartWithTwoProducts, {
        type:"DELETE_PRODUCT_FROM_CART",
        payload: 2
    })).toMatchSnapshot();
})
