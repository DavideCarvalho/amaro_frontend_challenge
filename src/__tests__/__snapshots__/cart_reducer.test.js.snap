// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add a second product 1`] = `
Object {
  "cart": Array [
    Object {
      "actual_price": "R$ 199,90",
      "code_color": "20002605_613",
      "color": "TAPE�ARIA",
      "color_slug": "tapecaria",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002605_615_catalog_1.jpg?1460136912",
      "installments": "3x R$ 66,63",
      "name": "VESTIDO TRANSPASSE BOW",
      "on_sale": false,
      "regular_price": "R$ 199,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "PP",
          "sku": "5807_343_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5807_343_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5807_343_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5807_343_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "5807_343_0_GG",
        },
      ],
      "style": "20002605",
    },
    Object {
      "actual_price": "R$ 99,90",
      "code_color": "20002570_614",
      "color": "PRETO",
      "color_slug": "preto",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002570_002_catalog_1.jpg?1459948578",
      "installments": "3x R$ 33,30",
      "name": "REGATA ALCINHA FOLK",
      "on_sale": false,
      "regular_price": "R$ 99,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "PP",
          "sku": "5723_40130843_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5723_40130843_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5723_40130843_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5723_40130843_0_G",
        },
        Object {
          "available": true,
          "size": "GG",
          "sku": "5723_40130843_0_GG",
        },
      ],
      "style": "20002570",
    },
  ],
}
`;

exports[`Adds new product 1`] = `
Object {
  "cart": Array [
    Object {
      "actual_price": "R$ 199,90",
      "code_color": "20002605_613",
      "color": "TAPE�ARIA",
      "color_slug": "tapecaria",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002605_615_catalog_1.jpg?1460136912",
      "installments": "3x R$ 66,63",
      "name": "VESTIDO TRANSPASSE BOW",
      "on_sale": false,
      "regular_price": "R$ 199,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "PP",
          "sku": "5807_343_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5807_343_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5807_343_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5807_343_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "5807_343_0_GG",
        },
      ],
      "style": "20002605",
    },
  ],
}
`;

exports[`Delete a existing product 1`] = `
Object {
  "cart": Array [
    Object {
      "actual_price": "R$ 199,90",
      "code_color": "20002605_613",
      "color": "TAPE�ARIA",
      "color_slug": "tapecaria",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002605_615_catalog_1.jpg?1460136912",
      "installments": "3x R$ 66,63",
      "name": "VESTIDO TRANSPASSE BOW",
      "on_sale": false,
      "regular_price": "R$ 199,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "PP",
          "sku": "5807_343_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5807_343_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5807_343_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5807_343_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "5807_343_0_GG",
        },
      ],
      "style": "20002605",
    },
  ],
}
`;

exports[`Delete a non existing product should return the same state 1`] = `
Object {
  "cart": Array [
    Object {
      "actual_price": "R$ 199,90",
      "code_color": "20002605_613",
      "color": "TAPE�ARIA",
      "color_slug": "tapecaria",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002605_615_catalog_1.jpg?1460136912",
      "installments": "3x R$ 66,63",
      "name": "VESTIDO TRANSPASSE BOW",
      "on_sale": false,
      "regular_price": "R$ 199,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "PP",
          "sku": "5807_343_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5807_343_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5807_343_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5807_343_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "5807_343_0_GG",
        },
      ],
      "style": "20002605",
    },
    Object {
      "actual_price": "R$ 99,90",
      "code_color": "20002570_614",
      "color": "PRETO",
      "color_slug": "preto",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002570_002_catalog_1.jpg?1459948578",
      "installments": "3x R$ 33,30",
      "name": "REGATA ALCINHA FOLK",
      "on_sale": false,
      "regular_price": "R$ 99,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "PP",
          "sku": "5723_40130843_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5723_40130843_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5723_40130843_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5723_40130843_0_G",
        },
        Object {
          "available": true,
          "size": "GG",
          "sku": "5723_40130843_0_GG",
        },
      ],
      "style": "20002570",
    },
  ],
}
`;

exports[`Returns default state 1`] = `
Object {
  "cart": Array [],
}
`;
